AppDelegate.swift

```diff
@@ -1,141 +1,378 @@
 import UIKit
 import CoreData
-import Firebase
-import FirebaseAuth
-import FirebaseRemoteConfig
 import CoreMotion
+import KeychainSwift
+
+func DLog(_ message: String, file:NSString = #file, line: Int = #line, functionName: String = #function) {
+    #if DEBUG
+    let formatter = DateFormatter()
+    formatter.dateFormat = "yyyy-MM-dd HH:mm:ss.SSS "
+    print("[\(formatter.string(from: Date()))][\(file.lastPathComponent):\(line)][\(functionName)]: \(message)")
+    #endif
+}
 
 @UIApplicationMain
 class AppDelegate: UIResponder, UIApplicationDelegate {
     var window: UIWindow?
-    var pogoMM: PogoMotionManager!
-
+    var motionManager : CMMotionManager!
+    var backgroundTask: UIBackgroundTaskIdentifier = .invalid // this is a task to clear data when going to background
+    
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-
-        // Override point for customization after application launch.
-        FirebaseApp.configure()
-
-        //configure the database manager
-        self.configureDatabaseManager()
-
-        //the below can be in a single configure method inside the BluetraceManager
+        setupCoredataDir()
+        Encounter.timestamp(for: .appStarted)
+        let firstRun = UserDefaults.standard.bool(forKey: "HasBeenLaunched")
+        if( !firstRun ) {
+            let keychain = KeychainSwift()
+            keychain.clear()
+            UserDefaults.standard.set(true, forKey: "HasBeenLaunched")
+        }
+        
+        let hasUserConsent = true
+        let hasUserCompletedOnboarding = UserDefaults.standard.bool(forKey: "turnedOnBluetooth")
         let bluetoothAuthorised = BluetraceManager.shared.isBluetoothAuthorized()
-        if  OnboardingManager.shared.completedBluetoothOnboarding && bluetoothAuthorised {
+        if (hasUserConsent && hasUserCompletedOnboarding && bluetoothAuthorised) {
```

Comment:
Hardcoded consent check?

Reply:
potentially were going to add a consent screen

Reply:
Haha, yeah, I noticed that too...